### PR TITLE
chore(weave): Drawer should push content not hover over

### PR DESCRIPTION
**Pull request by jwlee64**

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/useDrawerResize.tsx [hunk 1]
```diff
@@ -6,41 +6,41 @@ import {useFlexDirection} from './useFlexDirection';
 
 const setDrawerSize = (
   newSize: number,
-  setSize: (value: string) => void,
```
Comment (jwlee64) [inline weave-js/src/components/PagePanelComponents/Home/useDrawerResize.tsx:9]: converts this file to use and store the percent number not string

File: weave-js/src/components/PagePanelComponents/Home/useDrawerResize.tsx [hunk 2]
```diff
@@ -82,17 +82,11 @@ export const useDrawerResize = () => {
       e.preventDefault();
       if (flexDirection === 'row') {
         const newWidth =
-          ((document.body.offsetWidth -
```
Comment (jwlee64) [inline weave-js/src/components/PagePanelComponents/Home/useDrawerResize.tsx:85]: cleans up this calculation to do the same thing simpler

File: weave-js/src/components/PagePanelComponents/Home/Browse3.tsx [hunk 3]
```diff
@@ -333,35 +337,39 @@ const MainPeekingLayout: FC = () => {
...
-            margin: flexDirection === 'row' ? '60px 0 0 0' : '0 0 0 57px',
-            boxShadow:
-              flexDirection === 'row'
-                ? 'rgba(15, 15, 15, 0.04) 0px 0px 0px 1px, rgba(15, 15, 15, 0.03) 0px 3px 6px, rgba(15, 15, 15, 0.06) 0px 9px 24px'
-                : 'rgba(15, 15, 15, 0.04) 0px 0px 0px 1px, rgba(15, 15, 15, 0.03) 3px 0px 6px, rgba(15, 15, 15, 0.06) 9px 0px 24px',
-            borderLeft:
-              flexDirection === 'row' ? `1px solid ${MOON_200}` : 'none',
-            borderTop:
-              flexDirection === 'column' ? `1px solid ${MOON_200}` : 'none',
+            width: isFlexRow ? `${drawerWidthPct}%` : 'calc(100% - 57px)',
```
Comment (jwlee64) [inline weave-js/src/components/PagePanelComponents/Home/Browse3.tsx:365]: this just changes all the flexDirection == 'row' or 'column' to use a var

File: weave-js/src/components/PagePanelComponents/Home/Browse3.tsx [hunk 4]
```diff
@@ -333,35 +337,39 @@ const MainPeekingLayout: FC = () => {
...
+          transition: !isDrawerOpen
+            ? 'margin 225ms cubic-bezier(0, 0, 0.2, 1) 0ms'
+            : 'none',
+          marginRight:
+            !isDrawerOpen || !isFlexRow
+              ? 0
+              : // this is a hack to make the drawer width percentage work with the siderbar of 56px
+                `${drawerWidthPct / (1 - 56 / windowSize.width)}%`,
+          // this is vh because margin-bottom percentages are relative to the viewport width
+          marginBottom: !isDrawerOpen || isFlexRow ? 0 : `${drawerHeightPct}vh`,
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3.tsx:350]: curious why we need to account for the 56 pixels in the sidebar, but not the header bar

Comment (jwlee64) [reply weave-js/src/components/PagePanelComponents/Home/Browse3.tsx:350]: me too, its really weird, i couldnt figure it out

File: weave-js/src/components/PagePanelComponents/Home/Browse3.tsx [hunk 5]
```diff
@@ -333,35 +337,39 @@ const MainPeekingLayout: FC = () => {
...
           display: 'flex',
+          // This transition is from the mui drawer component, to keep the main content animation in similar
+          transition: !isDrawerOpen
+            ? 'margin 225ms cubic-bezier(0, 0, 0.2, 1) 0ms'
+            : 'none',
+          marginRight:
+            !isDrawerOpen || !isFlexRow
+              ? 0
+              : // this is a hack to make the drawer width percentage work with the siderbar of 56px
+                `${drawerWidthPct / (1 - 56 / windowSize.width)}%`,
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3.tsx:348]: This `56` and the `60` for the header feel like magic numbers that could change. Can we pull those into constants?

Note: it is sort of odd that this component "knows" about the space it is mounted in. Fine for now, but just calling out that it is an odd coupling between sibling/cousin components.

**Review by tssweeney — APPROVED** [review]
Yes! This looks and feels much better! Thank you (:

**Pull request merged**